### DISCUSSION TITLE
How can I have a panel with centred console markup?

### DISCUSSION THREAD

**Discussion by ciaransweet** [Textualize/textual#637] [Q&A] [answered]
If I render a Widget of:

```python
from rich.panel import Panel
# from rich.text import Text
from textual.widget import Widget


class TitleBar(Widget):

    def render(self) -> Panel:
        return Panel("[b]Terraform Workspaces[/b]\n:star: == Active Workspace", height=6)
```

I get:

![image](https://user-images.githubusercontent.com/9111975/183031380-c735439f-5b0d-48b0-a052-3806c28bc132.png)

But really I want the text centred in the panel. I tried to use `rich.Text` here:

```python
from rich.panel import Panel
from rich.text import Text
from textual.widget import Widget


class TitleBar(Widget):

    def render(self) -> Panel:
        return Panel(Text("[b]Terraform Workspaces[/b]\n:star: == Active Workspace", justify="center"), height=6)

```

I get:

![image](https://user-images.githubusercontent.com/9111975/183031669-116e0f65-d125-4cfb-8c91-bcf4bff70d36.png)

Which **is** centred... but my markup disappears 😅 

Appreciate any thoughts folks have! I'm having a great time playing with this

**Comment by willmcgugan** [comment, marked as answer]
Hi Ciaran,

You want `Text.from_markup()` to also process the console markup.

**Reply by ciaransweet** [reply]
Well that was easy 🤣 Cheers @willmcgugan!

**Reply by ciaransweet** [reply]
![image](https://user-images.githubusercontent.com/9111975/183034087-d92d965d-b65b-4373-9018-dfe6f162a703.png)
🎉
